src/components/FilmsList/FilmsList.js

```diff
@@ -1,19 +1,36 @@
-import React from 'react';
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import './FilmsList.scss';
 import { FilmCard } from '../FilmCard';
+import { store } from '../../store';
 
-export const FilmsList = (props) => {
-  const { films } = props;
+export class FilmsList extends Component {
+  state = {
+    films: store.getState().films,
+  }
 
-  return (
-    <div className="films">
-      {films.map(film => (
-        <FilmCard key={film.id} {...film} />
-      ))}
-    </div>
-  );
-};
+  componentDidMount() {
+    this.unsubscribe = store.subscribe(() => {
+      this.setState({ films: store.getState().films });
+    });
+  }
+
+  componentWillUnmount() {
+    this.unsubscribe();
+  }
+
+  render() {
+    const { films } = this.state;
+
+    return (
+      <div className="films">
+        {films.map(film => (
+          <FilmCard key={film.id} {...film} />
```

Comment:
what if I add the same movies? they will have the same id so key won't be unique

Reply:
now id is the same

Reply:
you need to create unique key